aws/data_source_aws_guardduty_detector.go

```diff
@@ -0,0 +1,95 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/guardduty"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func dataSourceAwsGuarddutyDetector() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceAwsGuarddutyDetectorRead,
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"enabled": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"service_role_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"finding_publishing_frequency": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func dataSourceAwsGuarddutyDetectorRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).guarddutyconn
+
+	detectorId := d.Get("id").(string)
+
+	if detectorId == "" {
+		log.Print("[DEBUG] No Guardduty ID has been defined, try to lookup using find (There should be only one)")
+		input := &guardduty.ListDetectorsInput{}
+
+		resp, err := conn.ListDetectors(input)
+		if err != nil {
+			return err
```

Comment:
To provide operators and code maintainers better context about errors, we should prefer to include information about when/why an error occurs, e.g.

```suggestion
			return fmt.Errorf("error listing GuardDuty Detectors: %s", err)
```

Reply:
Additional details added to the error output.